clang/test/Driver/sycl-no-rdc-fat-archive-win.cpp

```diff
@@ -5,14 +5,14 @@
 // RUN: echo "void foo(void) {}" > %t1.cpp
 // RUN: %clangxx -target x86_64-pc-windows-msvc -fsycl %t1.cpp -c -o %t1_bundle.o
 // RUN: llvm-ar cr %t_lib.a %t1_bundle.o
-// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=off --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=auto --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_kernel --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_source --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang_cl -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=off --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang_cl -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=auto --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang_cl -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_kernel --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
-// RUN: %clang_cl -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_source --sysroot=%S/Inputs/SYCL %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
+// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=off --sysroot=%S/Inputs/SYCL-windows %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
+// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=auto --sysroot=%S/Inputs/SYCL-windows %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
+// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_kernel --sysroot=%S/Inputs/SYCL-windows %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
+// RUN: %clang -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=per_source --sysroot=%S/Inputs/SYCL-windows %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
+// RUN: %clang_cl -### -fsycl -fno-sycl-rdc -fsycl-device-code-split=off /clang:--sysroot=%S/Inputs/SYCL-windows %t_lib.a 2>&1 -ccc-print-phases | FileCheck %s
```

Comment:
Instead of force feeding `--sysroot` does `/winsysroot` provide better way of setting up the env?

Reply:
Thanks for the review. It doesn't look like it works. Below is the output.

Intel/llvm without these changes (fails)
```
bash-3.2$ "d:\iusers\nsarnie\llvm\build\bin\clang.exe" "--driver-mode=cl" "-###" "-fsycl" "-fno-sycl-rdc" "--sysroot=D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp" "-ccc-print-phases"
clang: warning: unknown argument ignored in clang-cl: '--sysroot=D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows' [-Wunknown-argument]
clang version 16.0.0 (https://github.com/sarnex/llvm.git 7236bd9218c973d3a384b50f3327c9123c8a620e)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: d:\iusers\nsarnie\llvm\build\bin
                     +- 0: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (host-sycl)
                  +- 1: append-footer, {0}, c++, (host-sycl)
               +- 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
               |     +- 3: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (device-sycl)
               |  +- 4: preprocessor, {3}, c++-cpp-output, (device-sycl)
               |- 5: compiler, {4}, ir, (device-sycl)
            +- 6: offload, "host-sycl (x86_64-pc-windows-msvc)" {2}, "device-sycl (spir64-unknown-unknown)" {5}, c++-cpp-output
         +- 7: compiler, {6}, ir, (host-sycl)
      +- 8: backend, {7}, assembler, (host-sycl)
   +- 9: assembler, {8}, object, (host-sycl)
   |                 +- 10: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (host-sycl)
   |              +- 11: append-footer, {10}, c++, (host-sycl)
   |           +- 12: preprocessor, {11}, c++-cpp-output, (host-sycl)
   |           |     +- 13: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (device-sycl)
   |           |  +- 14: preprocessor, {13}, c++-cpp-output, (device-sycl)
   |           |- 15: compiler, {14}, ir, (device-sycl)
   |        +- 16: offload, "host-sycl (x86_64-pc-windows-msvc)" {12}, "device-sycl (spir64-unknown-unknown)" {15}, c++-cpp-output
   |     +- 17: compiler, {16}, ir, (host-sycl)
   |  +- 18: backend, {17}, assembler, (host-sycl)
   |- 19: assembler, {18}, object, (host-sycl)
+- 20: linker, {9, 19}, image, (host-sycl)
|     +- 21: sycl-post-link, {5}, tempfiletable, (device-sycl)
|     |  +- 22: file-table-tform, {21}, tempfilelist, (device-sycl)
|     |- 23: llvm-spirv, {22}, tempfilelist, (device-sycl)
|  +- 24: file-table-tform, {21, 23}, tempfiletable, (device-sycl)
|- 25: clang-offload-wrapper, {24}, object, (device-sycl)
|     +- 26: sycl-post-link, {15}, tempfiletable, (device-sycl)
|     |  +- 27: file-table-tform, {26}, tempfilelist, (device-sycl)
|     |- 28: llvm-spirv, {27}, tempfilelist, (device-sycl)
|  +- 29: file-table-tform, {26, 28}, tempfiletable, (device-sycl)
|- 30: clang-offload-wrapper, {29}, object, (device-sycl)
31: offload, "host-sycl (x86_64-pc-windows-msvc)" {20}, "device-sycl (spir64-unknown-unknown)" {25}, "device-sycl (spir64-unknown-unknown)" {30}, image
bash-3.2$
```

With change in this PR:
```
bash-3.2$ "d:\iusers\nsarnie\llvm\build\bin\clang.exe" "--driver-mode=cl" "-###" "-fsycl" "-fno-sycl-rdc" "/clang:--sysroot=D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp" "-ccc-print-phases"
clang version 16.0.0 (https://github.com/sarnex/llvm.git 7236bd9218c973d3a384b50f3327c9123c8a620e)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: d:\iusers\nsarnie\llvm\build\bin
                     +- 0: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (host-sycl)
                  +- 1: append-footer, {0}, c++, (host-sycl)
               +- 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
               |     +- 3: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (device-sycl)
               |  +- 4: preprocessor, {3}, c++-cpp-output, (device-sycl)
               |- 5: compiler, {4}, ir, (device-sycl)
            +- 6: offload, "host-sycl (x86_64-pc-windows-msvc)" {2}, "device-sycl (spir64-unknown-unknown)" {5}, c++-cpp-output
         +- 7: compiler, {6}, ir, (host-sycl)
      +- 8: backend, {7}, assembler, (host-sycl)
   +- 9: assembler, {8}, object, (host-sycl)
   |                 +- 10: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (host-sycl)
   |              +- 11: append-footer, {10}, c++, (host-sycl)
   |           +- 12: preprocessor, {11}, c++-cpp-output, (host-sycl)
   |           |     +- 13: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (device-sycl)
   |           |  +- 14: preprocessor, {13}, c++-cpp-output, (device-sycl)
   |           |- 15: compiler, {14}, ir, (device-sycl)
   |        +- 16: offload, "host-sycl (x86_64-pc-windows-msvc)" {12}, "device-sycl (spir64-unknown-unknown)" {15}, c++-cpp-output
   |     +- 17: compiler, {16}, ir, (host-sycl)
   |  +- 18: backend, {17}, assembler, (host-sycl)
   |- 19: assembler, {18}, object, (host-sycl)
+- 20: linker, {9, 19}, image, (host-sycl)
|           |        +- 21: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-crt.obj", object
|           |     +- 22: clang-offload-unbundler, {21}, object
|           |  +- 23: offload, " (spir64-unknown-unknown)" {22}, object
|           |  |     +- 24: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-complex.obj", object
|           |  |  +- 25: clang-offload-unbundler, {24}, object
|           |  |- 26: offload, " (spir64-unknown-unknown)" {25}, object
|           |  |     +- 27: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-complex-fp64.obj", object
|           |  |  +- 28: clang-offload-unbundler, {27}, object
|           |  |- 29: offload, " (spir64-unknown-unknown)" {28}, object
|           |  |     +- 30: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-cmath.obj", object
|           |  |  +- 31: clang-offload-unbundler, {30}, object
|           |  |- 32: offload, " (spir64-unknown-unknown)" {31}, object
|           |  |     +- 33: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-cmath-fp64.obj", object
|           |  |  +- 34: clang-offload-unbundler, {33}, object
|           |  |- 35: offload, " (spir64-unknown-unknown)" {34}, object
|           |  |     +- 36: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-msvc-math.obj", object
|           |  |  +- 37: clang-offload-unbundler, {36}, object
|           |  |- 38: offload, " (spir64-unknown-unknown)" {37}, object
|           |  |     +- 39: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-imf.obj", object
|           |  |  +- 40: clang-offload-unbundler, {39}, object
|           |  |- 41: offload, " (spir64-unknown-unknown)" {40}, object
|           |  |     +- 42: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-imf-fp64.obj", object
|           |  |  +- 43: clang-offload-unbundler, {42}, object
|           |  |- 44: offload, " (spir64-unknown-unknown)" {43}, object
|           |  |     +- 45: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-imf-bf16.obj", object
|           |  |  +- 46: clang-offload-unbundler, {45}, object
|           |  |- 47: offload, " (spir64-unknown-unknown)" {46}, object
|           |  |     +- 48: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-cassert.obj", object
|           |  |  +- 49: clang-offload-unbundler, {48}, object
|           |  |- 50: offload, " (spir64-unknown-unknown)" {49}, object
|           |  |     +- 51: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-cstring.obj", object
|           |  |  +- 52: clang-offload-unbundler, {51}, object
|           |  |- 53: offload, " (spir64-unknown-unknown)" {52}, object
|           |  |     +- 54: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-complex.obj", object
|           |  |  +- 55: clang-offload-unbundler, {54}, object
|           |  |- 56: offload, " (spir64-unknown-unknown)" {55}, object
|           |  |     +- 57: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-complex-fp64.obj", object
|           |  |  +- 58: clang-offload-unbundler, {57}, object
|           |  |- 59: offload, " (spir64-unknown-unknown)" {58}, object
|           |  |     +- 60: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-cmath.obj", object
|           |  |  +- 61: clang-offload-unbundler, {60}, object
|           |  |- 62: offload, " (spir64-unknown-unknown)" {61}, object
|           |  |     +- 63: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-cmath-fp64.obj", object
|           |  |  +- 64: clang-offload-unbundler, {63}, object
|           |  |- 65: offload, " (spir64-unknown-unknown)" {64}, object
|           |  |     +- 66: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-imf.obj", object
|           |  |  +- 67: clang-offload-unbundler, {66}, object
|           |  |- 68: offload, " (spir64-unknown-unknown)" {67}, object
|           |  |     +- 69: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-imf-fp64.obj", object
|           |  |  +- 70: clang-offload-unbundler, {69}, object
|           |  |- 71: offload, " (spir64-unknown-unknown)" {70}, object
|           |  |     +- 72: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-fallback-imf-bf16.obj", object
|           |  |  +- 73: clang-offload-unbundler, {72}, object
|           |  |- 74: offload, " (spir64-unknown-unknown)" {73}, object
|           |  |     +- 75: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-itt-user-wrappers.obj", object
|           |  |  +- 76: clang-offload-unbundler, {75}, object
|           |  |- 77: offload, " (spir64-unknown-unknown)" {76}, object
|           |  |     +- 78: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-itt-compiler-wrappers.obj", object
|           |  |  +- 79: clang-offload-unbundler, {78}, object
|           |  |- 80: offload, " (spir64-unknown-unknown)" {79}, object
|           |  |     +- 81: input, "D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows/lib\libsycl-itt-stubs.obj", object
|           |  |  +- 82: clang-offload-unbundler, {81}, object
|           |  |- 83: offload, " (spir64-unknown-unknown)" {82}, object
|           |- 84: linker, {23, 26, 29, 32, 35, 38, 41, 44, 47, 50, 53, 56, 59, 62, 65, 68, 71, 74, 77, 80, 83}, ir, (device-sycl)
|        +- 85: linker, {5, 84}, ir, (device-sycl)
|     +- 86: sycl-post-link, {85}, tempfiletable, (device-sycl)
|     |  +- 87: file-table-tform, {86}, tempfilelist, (device-sycl)
|     |- 88: llvm-spirv, {87}, tempfilelist, (device-sycl)
|  +- 89: file-table-tform, {86, 88}, tempfiletable, (device-sycl)
|- 90: clang-offload-wrapper, {89}, object, (device-sycl)
|        +- 91: linker, {15, 84}, ir, (device-sycl)
|     +- 92: sycl-post-link, {91}, tempfiletable, (device-sycl)
|     |  +- 93: file-table-tform, {92}, tempfilelist, (device-sycl)
|     |- 94: llvm-spirv, {93}, tempfilelist, (device-sycl)
|  +- 95: file-table-tform, {92, 94}, tempfiletable, (device-sycl)
|- 96: clang-offload-wrapper, {95}, object, (device-sycl)
97: offload, "host-sycl (x86_64-pc-windows-msvc)" {20}, "device-sycl (spir64-unknown-unknown)" {90}, "device-sycl (spir64-unknown-unknown)" {96}, image
bash-3.2$
```

With `/winsysroot'
```
bash-3.2$ "d:\iusers\nsarnie\llvm\build\bin\clang.exe" "--driver-mode=cl" "-###" "-fsycl" "-fno-sycl-rdc" "/winsysroot:D:\iusers\nsarnie\llvm\clang\test\Driver/Inputs/SYCL-windows" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp" "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp" "-ccc-print-phases"
clang version 16.0.0 (https://github.com/sarnex/llvm.git 7236bd9218c973d3a384b50f3327c9123c8a620e)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: d:\iusers\nsarnie\llvm\build\bin
                     +- 0: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (host-sycl)
                  +- 1: append-footer, {0}, c++, (host-sycl)
               +- 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
               |     +- 3: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp1.cpp", c++, (device-sycl)
               |  +- 4: preprocessor, {3}, c++-cpp-output, (device-sycl)
               |- 5: compiler, {4}, ir, (device-sycl)
            +- 6: offload, "host-sycl (x86_64-pc-windows-msvc)" {2}, "device-sycl (spir64-unknown-unknown)" {5}, c++-cpp-output
         +- 7: compiler, {6}, ir, (host-sycl)
      +- 8: backend, {7}, assembler, (host-sycl)
   +- 9: assembler, {8}, object, (host-sycl)
   |                 +- 10: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (host-sycl)
   |              +- 11: append-footer, {10}, c++, (host-sycl)
   |           +- 12: preprocessor, {11}, c++-cpp-output, (host-sycl)
   |           |     +- 13: input, "D:\iusers\nsarnie\llvm\build\tools\clang\test\Driver\Output\sycl-no-rdc-win.cpp.tmp2.cpp", c++, (device-sycl)
   |           |  +- 14: preprocessor, {13}, c++-cpp-output, (device-sycl)
   |           |- 15: compiler, {14}, ir, (device-sycl)
   |        +- 16: offload, "host-sycl (x86_64-pc-windows-msvc)" {12}, "device-sycl (spir64-unknown-unknown)" {15}, c++-cpp-output
   |     +- 17: compiler, {16}, ir, (host-sycl)
   |  +- 18: backend, {17}, assembler, (host-sycl)
   |- 19: assembler, {18}, object, (host-sycl)
+- 20: linker, {9, 19}, image, (host-sycl)
|     +- 21: sycl-post-link, {5}, tempfiletable, (device-sycl)
|     |  +- 22: file-table-tform, {21}, tempfilelist, (device-sycl)
|     |- 23: llvm-spirv, {22}, tempfilelist, (device-sycl)
|  +- 24: file-table-tform, {21, 23}, tempfiletable, (device-sycl)
|- 25: clang-offload-wrapper, {24}, object, (device-sycl)
|     +- 26: sycl-post-link, {15}, tempfiletable, (device-sycl)
|     |  +- 27: file-table-tform, {26}, tempfilelist, (device-sycl)
|     |- 28: llvm-spirv, {27}, tempfilelist, (device-sycl)
|  +- 29: file-table-tform, {26, 28}, tempfiletable, (device-sycl)
|- 30: clang-offload-wrapper, {29}, object, (device-sycl)
31: offload, "host-sycl (x86_64-pc-windows-msvc)" {20}, "device-sycl (spir64-unknown-unknown)" {25}, "device-sycl (spir64-unknown-unknown)" {30}, image
bash-3.2$
```